[ci] R-package jobs fail compiling {fs}: "cmake: command not found"

## Description

Linux R-package CI jobs are failing like this:

```text
** libs
using C++ compiler: ‘Ubuntu clang version 16.0.6 (++20231112100510+7cbf1a259152-1~exp1~20231112100554.106)’
make[1]: Entering directory '/tmp/RtmphHq33B/R.INSTALL1c6be873114/fs/src'
tar xf ../tools/libuv-v1.52.0.tar.gz && \
CC="clang-16" CFLAGS="-O3 -Wall -pedantic -fpic" CPPFLAGS="-I/usr/local/include" AR="ar" RANLIB="ranlib" LDFLAGS="-L/usr/local/lib" cmake -S libuv-v1.52.0 -B libuv-v1.52.0 -DLIBUV_BUILD_SHARED=OFF && cmake --build libuv-v1.52.0 && \
cp -f libuv-v1.52.0/libuv.a libuv && \
cp -Rf libuv-v1.52.0/include/* . &&\
rm -Rf libuv-v1.52.0
/bin/bash: line 2: cmake: command not found
make[1]: *** [Makevars:44: libuv] Error 127
make[1]: Leaving directory '/tmp/RtmphHq33B/R.INSTALL1c6be873114/fs/src'
ERROR: compilation failed for package ‘fs’
```

([build link](https://github.com/lightgbm-org/LightGBM/actions/runs/23417357866/job/68119575708?pr=7167#step:6:1616))

This is probably a result of the new release `{fs}` 2.0.0 that went up on March 6th. I see similar errors for several CRAN check flavors: https://cran.r-project.org/web/checks/check_results_fs.html

I strongly suspect CRAN has already sent an email to those maintainers and that they'll submit a new version with a fix soon.

## Notes

N/A

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -288,6 +288,7 @@ jobs:
           if type -f apt 2>&1 > /dev/null; then
             apt-get update
             apt-get install --no-install-recommends -y \
+                cmake \
                 devscripts \
                 texinfo \
                 texlive-latex-extra \
@@ -299,6 +300,7 @@ jobs:
           else
             yum update -y
             yum install -y \
+                cmake \
                 devscripts \
                 qpdf \
                 texinfo \

